### PR TITLE
Avoid overflow in IntervalYearMonthType::valueToString

### PR DESCRIPTION
**Pull request by marin-ma**

### CODE REVIEW

**Comment by marin-ma** [comment]
@mbasmanova Could you help to review? Thanks!

cc @kewang1024

File: velox/type/Type.h [hunk 1]
```diff
@@ -2088,6 +2088,10 @@ struct CppToType<Timestamp> : public CppToTypeBase<TypeKind::TIMESTAMP> {};
 template <>
 struct CppToType<Date> : public CppToTypeBase<TypeKind::INTEGER> {};
 
+template <>
+struct CppToType<IntervalYearMonthType>
```
Comment (mbasmanova) [inline velox/type/Type.h:2092]: Why is this needed?

Comment (marin-ma) [reply velox/type/Type.h:2092]: Without this change, it will fail to register a simple function with IntervalYearMonthType in its signature.

Comment (mbasmanova) [reply velox/type/Type.h:2092]: That's strange. This PR doesn't register any function.

Comment (marin-ma) [reply velox/type/Type.h:2092]: This PR is extracted from https://github.com/facebookincubator/velox/pull/8887 which adds `make_ym_interval` function with return type of `IntervalYearMonthType`
I'll remove this change as it's irrelevant to this PR.

**Review by mbasmanova — APPROVED** [review]
Thanks.

**Review by xiaoxmeng — APPROVED** [review]
@marin-ma thanks!

**Comment by mbasmanova** [comment]
@marin-ma Would you rebase?

**Review by kewang1024 — APPROVED** [review]
